PVTPackage/source/MultiphaseSystem/MultiphaseSystem.cpp

```diff
@@ -14,21 +14,210 @@
+ * @param variation Variation solution.
+ * @param delta Finite difference step.
+ * @param store Callback that will be called with the computed derivatives.
+ *
+ * Generic functions (pure code factorisation) that evaluates the derivatives
+ * through finite differences process between reference @p sysProps and variation @p diffed.
+ * This function then calls @p consumer, which role should mainly to store the computed data where they should be.
+ */
+template< class C >
+void updateDerivativeFiniteDifference( const FactorMultiphaseSystemProperties & reference,
+                                       const FactorMultiphaseSystemProperties & variation,
+                                       double delta,
+                                       const C & store )
+{
+  for( const pvt::PHASE_TYPE & phase : reference.getPhases() )
+  {
+    const double dPhaseMoleFraction = ( variation.getPhaseMoleFraction( phase ).value - reference.getPhaseMoleFraction( phase ).value ) / delta;
+    const double dMolecularWeight = ( variation.getMolecularWeight( phase ).value - reference.getMolecularWeight( phase ).value ) / delta;
+    const double dMoleDensity = ( variation.getMoleDensity( phase ).value - reference.getMoleDensity( phase ).value ) / delta;
+    const double dMassDensity = ( variation.getMassDensity( phase ).value - reference.getMassDensity( phase ).value ) / delta;
+
+    std::vector< double > dMoleComposition;
+    {
+      const std::vector< double > & refMoleComposition = reference.getMoleComposition( phase ).value;
+      const std::vector< double > & varMoleComposition = variation.getMoleComposition( phase ).value;
+      dMoleComposition.resize( refMoleComposition.size() );
+      for( std::size_t i = 0; i < refMoleComposition.size(); ++i )
+      {
+        dMoleComposition[i] = ( varMoleComposition[i] - refMoleComposition[i] ) / delta;
+      }
+    }
+
+    store( phase, dPhaseMoleFraction, dMolecularWeight, dMoleDensity, dMassDensity, dMoleComposition );
+  }
+}
+
+void MultiphaseSystem::updateDerivativeDPFiniteDifference( FactorMultiphaseSystemProperties & sysProps,
+                                                           const FactorMultiphaseSystemProperties & diffed,
+                                                           double dPressure )
+{
+  auto callBack = [&sysProps]( pvt::PHASE_TYPE phase,
+                               double dPhaseMoleFraction,
+                               double dMolecularWeight,
+                               double dMoleDensity,
+                               double dMassDensity,
+                               const std::vector< double > & dMoleComposition )
+  {
+    sysProps.setPhaseMoleFractionDP( phase, dPhaseMoleFraction );
+    sysProps.setMolecularWeightDP( phase, dMolecularWeight );
+    sysProps.setMoleDensityDP( phase, dMoleDensity );
+    sysProps.setMassDensityDP( phase, dMassDensity );
+    sysProps.setMoleCompositionDP( phase, dMoleComposition );
```

Comment:
OK; is there already something defined in our coding rules for GEOSX?
Or do we have already adopted this rule (for function names I mean)?

Reply:
This syntax with the differential operator written as "var1_dVar2" was verbally agreed upon during the dev meeting.

Reply:
IIUC `dVar1_dVar2` was about variables, not setters/getters, function names.
Do we want `set_dMoleComposition_dP`, `setDMoleComposition_dP`, `setMoleComposition_dP` (to keep symmetry with `setMoleComposition`, but no initial `d`)?

Reply:
Hello Thomas, thanks for going over all my (nitpicking) comments. So here, I would suggest to stick to the convention used by @klevzoff for the getters of the other fluid models. For example: 
https://github.com/GEOSX/GEOSX/blob/396f50bea86a75b0bfbbfcc089b9d33918f9e14f/src/coreComponents/constitutive/fluid/MultiFluidBase.hpp#L258-L267 
I personally like this convention, but I am not sure others agree. I am going to write this down in PR https://github.com/GEOSX/GEOSX/pull/1187 (I forgot to do that, sorry).

Reply:
But everywhere else in the code you have `get` for getters and `set` for setters.
Do you want to bypass this other rule?

Reply:
Yes you are right I did not think about this other convention (`get` and `set`) that's a good point. I think here it makes sense to keep `set` and `get` because you do both, whereas in the example I was mentioning there is no need for a setter. The convention that I was referring to is the `dVar1_dVar2` that makes code more readable when writing derivatives.